### ISSUE TITLE
DaskTaskRunner user logs aren't formatted with timestamps (like the other logs)

### ISSUE THREAD

**Issue by ahuang11** [PrefectHQ/prefect-dask#10]
From https://discourse.prefect.io/t/how-to-use-logging-in-orion-a-tutorial-by-andrew-brookins/197/4?u=ahuang11

```
import os
import dask

os.environ.update(
    PREFECT_ORION_DATABASE_CONNECTION_TIMEOUT="60.0",
    PREFECT_LOGGING_EXTRA_LOGGERS="test1",
    PREFECT_API_URL="http://127.0.0.1:4200/api",
)

from prefect.flows import flow
from prefect.tasks import task

from prefect_dask.task_runners import DaskTaskRunner as runner

# from prefect.task_runners import SequentialTaskRunner as runner
# from prefect_ray.task_runners import RayTaskRunner as runner
from prefect.logging import get_logger
import logging


@flow(task_runner=runner())
def testflow():
    testtask()


@task
def testtask():
    log = get_logger("task_runner.dask")
    log.warning("warning hi")
    log.debug("debug hi")

    log = logging.getLogger()
    log.warning("warning std root log")
    log.debug("debug std root log")

    log = logging.getLogger("test1")
    log.warning("warning extra log")
    log.debug("debug extra log")


if __name__ == "__main__":
    testflow()
```

Outputs (highlighted with <HERE>):
```
10:07:25.570 | INFO    | prefect.task_runner.dask - The Dask dashboard is available at http://127.0.0.1:8787/status
10:07:25.595 | WARNING | Flow run 'true-jacamar' - No default storage is configured on the server. Results from this flow run will be stored in a temporary directory in its runtime environment.
10:07:25.656 | INFO    | Flow run 'true-jacamar' - Created task run 'testtask-e0d484d7-0' for task 'testtask'

<HERE>
warning hi
warning std root log
warning extra log
<HERE>

10:07:29.678 | INFO    | Flow run 'true-jacamar' - Finished in state Completed('All states completed.')
```

**Comment by madkinsz** [comment]
It looks like the `setup_logging` call in `begin_task_run` is not working.

**Comment by ahuang11** [comment]
Tips on how to get it resolved?

I think this may be related?
https://saturncloud.io/docs/examples/python/dask/special-topics/qs-special-topics-logging/

How does `setup_logging` work?

Do we need to attach an additional `distributed.admin` logger to https://github.com/PrefectHQ/orion/blob/main/src/prefect/settings.py#L339?

**Comment by ahuang11** [comment]
```
import logging
import dask
import dask.distributed

logger = logging.getLogger('distributed.worker')
logger.setLevel(10)

print(logger.level) # 10
print(logger.getEffectiveLevel()) # 10

@dask.delayed
def doSomething():
    print(logger.level) # 20
    print(logger.getEffectiveLevel()) # 20
    return 1

g = doSomething()

# i tested this by running the cluster somewhere else (not using LocalCluster)
c = dask.distributed.Client()
r = c.compute(g)

print(r.result())
```

Results in 
10
10

20
20
1

**Comment by ahuang11** [comment]
And if I set the logger name to `prefect.task_runners`, the logging format finally gets updated.
```
logging:
  version: 1
  admin:
    log-format: '%(name)s - %(levelname)s - %(message)s'
  formatters:
    custom:
      format: "DASK ADMIN %(asctime)s __ %(levelname)-7s __ %(name)s __ %(message)s"
      datefmt: "%H:%M:%S"
  handlers:
    console:
      formatter: custom
      class: logging.StreamHandler
      level: INFO
  loggers:
    prefect.task_runs:
      level: INFO
      handlers:
        - console
```

**Comment by ahuang11** [comment]
I tried updating the config for the workers:
```
import dask
import prefect
from prefect import flow, task, get_run_logger
from prefect.logging import get_logger
from prefect_dask import DaskTaskRunner


prefect.logging.configuration.setup_logging()
dask.config.set(logging={'admin': {'log-format': '%(name)s - %(levelname)s - %(message)s'},
             'formatters': {'custom': {'datefmt': '%H',
                                       'format': 'DASK DASK %(asctime)s __ '
                                                 '%(levelname)-7s __ %(name)s '
                                                 '__ %(message)s'}},
             'handlers': {'console': {'class': 'logging.StreamHandler',
                                      'formatter': 'custom',
                                      'level': 'INFO'}},
             'loggers': {'prefect.task_runs': {'handlers': ['console'],
                                               'level': 'INFO'}},
             'version': 1})

@task
def lazy_exponent(args):
    logger = get_run_logger()
    x, y = args
    result = x**y
    # the logging call to keep tabs on the computation
    logger.warning(f"Computed exponent {x}^{y} = {result}")
    return result


@flow(task_runner=DaskTaskRunner())
def test_flow():
    inputs = [[1, 2], [3, 4]]
    results = []
    for i in inputs:
        results.append(lazy_exponent(i))
    return results

test_flow().result()
```

However, it does not work due to https://github.com/dask/distributed/issues/3882

**Comment by ahuang11** [comment]
Another thought is copying `logging.yaml` into `~/.config/dask/prefect.yaml` on initiation.

**Comment by madkinsz** [comment]
It seems likely that `setup_logging` is not working as intended, that the call is being skipped. I'd investigate that before looking into Dask-specific solutions.

**Comment by ahuang11** [comment]
Can you elaborate on "that the call is being skipped"?

I added a print at the end, and it does show up.
Source:
```
def setup_logging() -> None:
    global PROCESS_LOGGING_CONFIG

    # If the user has specified a logging path and it exists we will ignore the
    # default entirely rather than dealing with complex merging
    config = load_logging_config(
        (
            PREFECT_LOGGING_SETTINGS_PATH.value()
            if PREFECT_LOGGING_SETTINGS_PATH.value().exists()
            else DEFAULT_LOGGING_SETTINGS_PATH
        )
    )

    if PROCESS_LOGGING_CONFIG:
        # Do not allow repeated configuration calls, only warn if the config differs
        if PROCESS_LOGGING_CONFIG != config:
            warnings.warn(
                "Logging can only be setup once per process, the new logging config "
                f"will be ignored.",
                stacklevel=2,
            )
        return

    logging.config.dictConfig(config)

    # Copy configuration of the 'prefect.extra' logger to the extra loggers
    extra_config = logging.getLogger("prefect.extra")

    for logger_name in PREFECT_LOGGING_EXTRA_LOGGERS.value():
        logger = logging.getLogger(logger_name)
        for handler in extra_config.handlers:
            logger.addHandler(handler)
            if logger.level == logging.NOTSET:
                logger.setLevel(extra_config.level)
            logger.propagate = extra_config.propagate

    PROCESS_LOGGING_CONFIG = config
    print(config)
```

Test:
```
from prefect import flow, task, get_run_logger
from prefect_dask.task_runners import DaskTaskRunner


@flow(task_runner=DaskTaskRunner())
def testflow():
    testtask()


@task
def testtask():
    log = get_run_logger()
    log.warning("warning hi")
    log.debug("debug hi")


if __name__ == "__main__":
    testflow()
```

Output:
```
{'version': 1, 'disable_existing_loggers': False, 'formatters': {'simple': {'format': '%(asctime)s.%(msecs)03d | %(message)s', 'datefmt': '%H:%M:%S'}, 'standard': {'format': '%(asctime)s.%(msecs)03d | %(levelname)-7s | %(name)s - %(message)s', 'datefmt': '%H:%M:%S'}, 'flow_runs': {'format': '%(asctime)s.%(msecs)03d | %(levelname)-7s | Flow run %(flow_run_name)r - %(message)s', 'datefmt': '%H:%M:%S'}, 'task_runs': {'format': '%(asctime)s.%(msecs)03d | %(levelname)-7s | Task run %(task_run_name)r - %(message)s', 'datefmt': '%H:%M:%S'}, 'json': {'class': 'prefect.logging.formatters.JsonFormatter', 'format': 'default'}}, 'handlers': {'console': {'level': 0, 'class': 'logging.StreamHandler', 'formatter': 'standard'}, 'console_flow_runs': {'level': 0, 'class': 'logging.StreamHandler', 'formatter': 'flow_runs'}, 'console_task_runs': {'level': 0, 'class': 'logging.StreamHandler', 'formatter': 'task_runs'}, 'orion': {'level': 0, 'class': 'prefect.logging.handlers.OrionHandler'}}, 'loggers': {'prefect': {'level': 'INFO', 'handlers': ['console'], 'propagate': False}, 'prefect.extra': {'level': 'INFO', 'handlers': ['orion', 'console'], 'propagate': False}, 'prefect.flow_runs': {'level': 'NOTSET', 'handlers': ['orion', 'console_flow_runs'], 'propagate': False}, 'prefect.task_runs': {'level': 'NOTSET', 'handlers': ['orion', 'console_task_runs'], 'propagate': False}, 'prefect.orion': {'level': 'WARNING'}, 'uvicorn': {'level': 'WARNING', 'handlers': ['console'], 'propagate': False}, 'fastapi': {'level': 'WARNING', 'handlers': ['console'], 'propagate': False}}, 'root': {'level': 'WARNING', 'handlers': ['console']}}
09:08:53.819 | INFO    | prefect.engine - Created flow run 'private-puffin' for flow 'testflow'
09:08:53.819 | INFO    | Flow run 'private-puffin' - Using task runner 'DaskTaskRunner'
09:08:53.822 | INFO    | prefect.task_runner.dask - Creating a new Dask cluster with `distributed.deploy.local.LocalCluster`
09:08:56.035 | INFO    | prefect.task_runner.dask - The Dask dashboard is available at http://127.0.0.1:8787/status
09:08:56.496 | INFO    | Flow run 'private-puffin' - Created task run 'testtask-e0d484d7-0' for task 'testtask'
warning hi
09:09:00.869 | INFO    | Flow run 'private-puffin' - Finished in state Completed('All states completed.')
```

Is setup_logging supposed to be instantiated multiple times?

**Comment by madkinsz** [comment]
`setup_logging` is designed to only run once per process, but the variable indicating that is has already run may be being copied over to the Dask worker or something preventing it from running there

**Comment by anna-geller** [comment]
just mentioning there is a related PR here https://github.com/dask/distributed/pull/7276

**Comment by tibuch** [comment]
Hi,
I just looked through the different issues and PRs regarding login from dask distributed and found that this [PR](https://github.com/dask/distributed/pull/7297) got merged 2 days ago. Would this have a positive impact on the Prefect side? 

I am hoping that this will resolve the logging issue I encounter here:
```
from prefect import flow, get_run_logger, task
from prefect_dask import DaskTaskRunner

@task()
def a_task(i):
    logger = get_run_logger()
    logger.info(f"I am logging from a task. {i}")

@flow(
    name="Logger Test",
    task_runner=DaskTaskRunner(),
)
def flow():
    logger = get_run_logger()
    logger.info("Flow starts!")

    a_task.map(range(3))

    logger.info("Flow ends!")
```

Which produces only the following output:
```
16:05:00.779 | INFO    | prefect.engine - Created flow run 'hairy-owl' for flow 'Logger Test'
16:05:00.779 | INFO    | prefect.task_runner.dask - Creating a new Dask cluster with `distributed.deploy.local.LocalCluster`
16:05:02.965 | INFO    | prefect.task_runner.dask - The Dask dashboard is available at http://127.0.0.1:8787/status
16:05:04.322 | INFO    | Flow run 'hairy-owl' - Flow starts!
16:05:04.327 | INFO    | Flow run 'hairy-owl' - Flow ends!
16:05:04.546 | INFO    | Flow run 'hairy-owl' - Created task run 'a_task-800f1d79-2' for task 'a_task'
16:05:04.930 | INFO    | Flow run 'hairy-owl' - Submitted task run 'a_task-800f1d79-2' for execution.
16:05:04.937 | INFO    | Flow run 'hairy-owl' - Created task run 'a_task-800f1d79-1' for task 'a_task'
16:05:04.946 | INFO    | Flow run 'hairy-owl' - Submitted task run 'a_task-800f1d79-1' for execution.
16:05:04.953 | INFO    | Flow run 'hairy-owl' - Created task run 'a_task-800f1d79-0' for task 'a_task'
16:05:04.958 | INFO    | Flow run 'hairy-owl' - Submitted task run 'a_task-800f1d79-0' for execution.
16:05:06.653 | INFO    | Flow run 'hairy-owl' - Finished in state Completed('All states completed.')
```

The expected output would be (obtained with the default task runner):
```
16:08:10.958 | INFO    | prefect.engine - Created flow run 'eccentric-fennec' for flow 'Logger Test'
16:08:12.168 | INFO    | Flow run 'eccentric-fennec' - Flow starts!
16:08:12.171 | INFO    | Flow run 'eccentric-fennec' - Flow ends!
16:08:12.374 | INFO    | Flow run 'eccentric-fennec' - Created task run 'a_task-800f1d79-2' for task 'a_task'
16:08:12.375 | INFO    | Flow run 'eccentric-fennec' - Submitted task run 'a_task-800f1d79-2' for execution.
16:08:12.380 | INFO    | Flow run 'eccentric-fennec' - Created task run 'a_task-800f1d79-1' for task 'a_task'
16:08:12.380 | INFO    | Flow run 'eccentric-fennec' - Submitted task run 'a_task-800f1d79-1' for execution.
16:08:12.384 | INFO    | Flow run 'eccentric-fennec' - Created task run 'a_task-800f1d79-0' for task 'a_task'
16:08:12.385 | INFO    | Flow run 'eccentric-fennec' - Submitted task run 'a_task-800f1d79-0' for execution.
16:08:12.956 | INFO    | Task run 'a_task-800f1d79-2' - I am logging from a task. 2
16:08:12.962 | INFO    | Task run 'a_task-800f1d79-0' - I am logging from a task. 0
16:08:13.164 | INFO    | Task run 'a_task-800f1d79-0' - Finished in state Completed()
16:08:13.169 | INFO    | Task run 'a_task-800f1d79-1' - I am logging from a task. 1
16:08:13.246 | INFO    | Task run 'a_task-800f1d79-2' - Finished in state Completed()
16:08:13.352 | INFO    | Task run 'a_task-800f1d79-1' - Finished in state Completed()
16:08:13.677 | INFO    | Flow run 'eccentric-fennec' - Finished in state Completed('All states completed.')
```

All the best and thank you for developing Prefect!

**Comment by ahuang11** [comment]
I think this is now fixed in latest Prefect
![image](https://user-images.githubusercontent.com/15331990/204896977-c1b9be74-e458-4696-9fdf-489f4004b327.png)
